l1-contracts/src/core/libraries/StakingQueue.sol

```diff
@@ -0,0 +1,58 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity >=0.8.27;
+
+import {Errors} from "./Errors.sol";
+
+struct DepositArgs {
+  address attester;
+  address withdrawer;
+  bool onCanonical;
+}
+
+struct StakingQueue {
+  mapping(uint256 index => DepositArgs validator) validators;
+  uint128 first;
+  uint128 last;
+}
+
+library StakingQueueLib {
+  function init(StakingQueue storage self) internal {
+    self.first = 1;
+    self.last = 1;
+  }
+
+  function enqueue(
+    StakingQueue storage self,
+    address _attester,
+    address _withdrawer,
+    bool _onCanonical
+  ) internal returns (uint256) {
+    uint256 queueLocation = self.last;
+
+    self.validators[queueLocation] =
+      DepositArgs({attester: _attester, withdrawer: _withdrawer, onCanonical: _onCanonical});
+    self.last += 1;
```

Comment:
fun fact, i put this into a small demo contract and replacing this with 
```suggestion
    self.last = uint128(queueLocation + 1);
```

removes 1 sload; if possible use cached storage values